### PR TITLE
[mlir][ODS] Omit printing default-valued attributes in oilists

### PR DESCRIPTION
**Pull request by MacDue**
This makes these match the behaviour of optional attributes (which are omitted when they are their default value of `none`). This allows for concise assembly formats without a custom printer.

An extra print of " " is also removed, this does change any existing uses of oilists, but if the parameter before the oilist is optional, that would previously add an extra space.

### CODE REVIEW

File: mlir/tools/mlir-tblgen/OpFormatGen.cpp [hunk 1]
```diff
@@ -2009,6 +2009,15 @@ static void genEnumAttrPrinter(const NamedAttribute *var, const Operator &op,
           "  }\n";
 }
 
+static void genNonDefaultValueCheck(MethodBody &body, const Operator &op,
```
Comment (c-rhodes) [inline mlir/tools/mlir-tblgen/OpFormatGen.cpp:2012]: please could you add a comment at the top with short-description

**Review by c-rhodes — APPROVED** [review]
this looks ok to me but I'm not very familiar with code so please allow time for others to review before landing

**Comment by kiranchandramohan** [comment]
Nice. OpenMP test changes look OK to me.

**Review by shraiysh — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**